pkg/primitives/vm/utils.go

```diff
@@ -103,6 +100,12 @@ func (p *Manager) prepContainer(
 	// via the flist, so we have control over when to decomission this volume.
 	// remounting in RW mode
 	volName := fmt.Sprintf("rootfs:%s", wl.ID.String())
+
+	isFirstBoot, err := storage.VolumeExists(ctx, volName)
```

Comment:
`isFirstBoot` is a weird name for that variable because, If true, it means it's NOT first boot, right ? So now instead of cleaning the keys on actually the first boo, the cleaning happens on any boot that is not the frist boot.

Instead i would name this `volumeExists` instead. It's relevant to the actual value that it holds.